client/Client.cpp

```diff
@@ -180,14 +180,15 @@ events::EventBus * CClient::eventBus() const
 	return clientEventBus.get();
 }
 
-void CClient::newGame()
+void CClient::newGame(CGameState * gameState)
 {
 	CSH->th->update();
 	CMapService mapService;
-	gs = new CGameState();
+	gs = gameState ? gameState : new CGameState();

```

Comment:
maybe name `gameState` something like `currentGameState` or `existingGameState`?

Reply:
Really it does not add any more meaning if it is that gameState parameter in load(). I can only suggest gameStateToLoad. Still looks weird. Why can not we initiate reloading of the same map?

Reply:
Naming - gameStateToLoad
Logic - Why we decided to pass gamestate via network? To avoid random generator desync issues?

Reply:
in regards of naming: newGame(gameState) is quite understandable that new game will be started using gamestate from the argument, isn't it?

Reply:
we're talking about method parameter name here, not about the calling side :)